Packs/GoogleChronicleBackstory/Integrations/GoogleChronicleBackstoryStreamingAPI/GoogleChronicleBackstoryStreamingAPI.py

```diff
@@ -0,0 +1,925 @@
+                    set_integration_context(integration_context)
+                    continue
+                user_rule_detections = []
+                chronicle_rule_detections = []
+                detection_identifiers = integration_context.get('detection_identifiers', [])
+                curatedrule_detection_identifiers = integration_context.get('curatedrule_detection_identifiers', [])
+
+                for raw_detection in detections:
+                    raw_detection_type = str(raw_detection.get('type', ''))
+                    if raw_detection_type.upper() == RULE_DETECTION:
+                        user_rule_detections.append(raw_detection)
+                    elif raw_detection_type.upper() == GCTI_FINDING:
+                        chronicle_rule_detections.append(raw_detection)
+
+                user_rule_detections = deduplicate_detections(user_rule_detections, detection_identifiers)
+                chronicle_rule_detections = deduplicate_curatedrule_detections(
+                    chronicle_rule_detections, curatedrule_detection_identifiers)
+                detection_to_process: list[dict] = []
+                add_detections_in_incident_list(user_rule_detections, detection_to_process)
+                detection_incidents: list[dict] = convert_events_to_actionable_incidents(detection_to_process)
+                curatedrule_detection_to_process: list[dict] = []
+                add_curatedrule_detections_in_incident_list(chronicle_rule_detections, curatedrule_detection_to_process)
+                curatedrule_incidents: list[dict] = convert_curatedrule_events_to_actionable_incidents(
+                    curatedrule_detection_to_process)
+                sample_events = detection_incidents[:5]
+                sample_events.extend(curatedrule_incidents[:5])
+                if sample_events:
+                    integration_context.update({'sample_events': json.dumps(sample_events)})
+                    set_integration_context(integration_context)
+                incidents = detection_incidents
+                incidents.extend(curatedrule_incidents)
+                integration_context.update({'continuation_time': continuation_time})
+                if not incidents:
+                    set_integration_context(integration_context)
+                    continue
+                total_ingested_incidents = 0
+                length_of_incidents = len(incidents)
+                while total_ingested_incidents < len(incidents):
+                    current_batch = IDEAL_BATCH_SIZE if (
+                        total_ingested_incidents + IDEAL_BATCH_SIZE <= length_of_incidents) else (
+                            length_of_incidents - total_ingested_incidents)
+                    demisto.debug(f"{CHRONICLE_STREAM_DETECTIONS} No. of detections being ingested: {current_batch}.")
+                    demisto.createIncidents(incidents[total_ingested_incidents: total_ingested_incidents + current_batch])
```

Comment:
I would have added a debug log for the `total_ingested_incidents` as well

Reply:
This could be done but we've kept logs batchwise log for more clarity.